src/tasks/gitea/gitea-drone-oauth.test.ts

```diff
@@ -0,0 +1,146 @@
+  beforeEach(() => {
+    sandbox = sinon.createSandbox()
+    consoleLog = sandbox.stub(console, 'log')
+  })
+  afterEach(() => {
+    sandbox.restore()
+  })
+  it('should successfully validate secret', () => {
+    const stubRemoteSecret: DroneSecret = {
+      clientId: Buffer.from('abc').toString('base64'),
+      clientSecret: Buffer.from('def').toString('base64'),
+    }
+    const stubOauthApps: OAuth2Application[] = [
+      {
+        name: 'otomi-drone',
+        clientId: 'abc',
+      },
+    ]
+    expect(isSecretValid(stubRemoteSecret, stubOauthApps)).to.be.true
+  })
+  it('should fail to validate on no secret', () => {
+    const stubOauthApps: OAuth2Application[] = [
+      {
+        name: 'otomi-drone',
+        clientId: 'abc',
+      },
+    ]
+    expect(isSecretValid(undefined, stubOauthApps)).to.be.false
+    expect(consoleLog.calledWith('Remote secret was not found')).to.be.true
+  })
+  it('should fail validation when the secret contains invalid data', () => {
+    const stubRemoteSecret: DroneSecret = {
+      clientId: '',
+      clientSecret: Buffer.from('def').toString('base64'),
+    }
+    const stubOauthApps: OAuth2Application[] = [
+      {
+        name: 'otomi-drone',
+        clientId: 'abc',
+      },
+    ]
+    expect(isSecretValid(stubRemoteSecret, stubOauthApps)).to.be.false
+    expect(consoleLog.calledWith('Remote secret values were empty')).to.be.true
+  })
+  it('should fail validation when the secret contains invalid data', () => {
+    const stubRemoteSecret: DroneSecret = {
+      clientId: Buffer.from('abc').toString('base64'),
+      clientSecret: '',
+    }
+    const stubOauthApps: OAuth2Application[] = [
+      {
+        name: 'otomi-drone',
+        clientId: 'abc',
+      },
+    ]
+    expect(isSecretValid(stubRemoteSecret, stubOauthApps)).to.be.false
+    expect(consoleLog.calledWith('Remote secret values were empty')).to.be.true
+  })
+  it('should fail to validate on empty OAuthApp list', () => {
+    const stubRemoteSecret: DroneSecret = {
+      clientId: Buffer.from('abc').toString('base64'),
+      clientSecret: Buffer.from('def').toString('base64'),
+    }
+    const stubOauthApps: OAuth2Application[] = []
+    expect(isSecretValid(stubRemoteSecret, stubOauthApps)).to.be.false
+    expect(consoleLog.calledWith("Gitea doesn't have any oauth application defined")).to.be.true
+  })
+  it('should fail to validate on non equal values', () => {
+    const stubRemoteSecret: DroneSecret = {
+      clientId: Buffer.from('abc').toString('base64'),
+      clientSecret: Buffer.from('def').toString('base64'),
+    }
+    const stubOauthApps: OAuth2Application[] = [
+      {
+        name: 'otomi-drone',
+        clientId: 'abcd',
+      },
+    ]
+    expect(isSecretValid(stubRemoteSecret, stubOauthApps)).to.be.false
+    expect(consoleLog.calledWith('OAuth data did not match with expected secret')).to.be.true
+  })
+})
+
+describe('Gitea-Drone: Parse CSRF Token', () => {
```

Comment:
should be removed entirely since you will use the force and install wheels from npm

Reply:
These tests are good unit test. 
What do you mean by saying `will use the force and install wheels from npm` ?

Reply:
Really? After all my comments you still advocate for useless tests?

Reply:
when I say use the force it means you don't come up with it yourself but instead use the powers given to you through open source software. Use micro libraries to avoid inventing a cookie parser.